devpack/src/main/java/io/neow3j/devpack/compiler/Compiler.java

```diff
@@ -2,224 +2,565 @@
+import io.neow3j.devpack.framework.annotations.Syscall.Syscalls;
+import io.neow3j.model.types.ContractParameterType;
+import io.neow3j.protocol.core.methods.response.NeoGetContractState.ContractState.ContractManifest;
+import io.neow3j.protocol.core.methods.response.NeoGetContractState.ContractState.ContractManifest.ContractABI;
+import io.neow3j.protocol.core.methods.response.NeoGetContractState.ContractState.ContractManifest.ContractABI.ContractEvent;
+import io.neow3j.protocol.core.methods.response.NeoGetContractState.ContractState.ContractManifest.ContractABI.ContractMethod;
+import io.neow3j.protocol.core.methods.response.NeoGetContractState.ContractState.ContractManifest.ContractFeatures;
+import io.neow3j.protocol.core.methods.response.NeoGetContractState.ContractState.ContractManifest.ContractGroup;
+import io.neow3j.protocol.core.methods.response.NeoGetContractState.ContractState.ContractManifest.ContractPermission;
 import io.neow3j.utils.BigIntegers;
 import io.neow3j.utils.Numeric;
-import java.io.ByteArrayOutputStream;
 import java.io.IOException;
 import java.math.BigInteger;
-import java.util.Iterator;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
 import java.util.List;
+import java.util.Map;
+import java.util.Optional;
 import org.objectweb.asm.ClassReader;
 import org.objectweb.asm.Type;
 import org.objectweb.asm.tree.AbstractInsnNode;
 import org.objectweb.asm.tree.AnnotationNode;
 import org.objectweb.asm.tree.ClassNode;
+import org.objectweb.asm.tree.IntInsnNode;
 import org.objectweb.asm.tree.LdcInsnNode;
+import org.objectweb.asm.tree.LocalVariableNode;
 import org.objectweb.asm.tree.MethodInsnNode;
 import org.objectweb.asm.tree.MethodNode;
 import org.objectweb.asm.tree.VarInsnNode;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 public class Compiler {
 
-    public static void main(String[] args) throws Exception {
-        ClassReader reader = new ClassReader("io.neow3j.devpack.template.HelloWorldContract");
+    private static final Logger log = LoggerFactory.getLogger(Compiler.class);
+    private static final ObjectMapper objectMapper = new ObjectMapper();
+
+    public static final int MAX_PARAMS_COUNT = 255;
+    public static final int MAX_LOCAL_VARIABLES_COUNT = 255;
+//    public static final int MAX_STATIC_FIELDS_COUNT = 255;
+
+    private int currentNeoAddr;
+
+    /**
+     * Compiles the class with the given name to NeoVM code.
+     *
+     * @param name the fully qualified name of the class.
+     */
+    public CompilationResult compileClass(String name) throws IOException {
+        ClassReader reader = new ClassReader(name);
         ClassNode n = new ClassNode();
         reader.accept(n, 0);
+        MethodNode entryPoint = getEntryPoint(n);
+        NeoMethod neoMethod = handleMethod(entryPoint);
+        byte[] script = neoMethod.toByteArray();
+        NefFile nef = new NefFile("neow3j", new Version(3, 0, 0, 0), script);
```

Comment:
about version and other params: same comment as I mentioned before... if there are static values, which will not change much, add to a constant class with static values.

Reply:
I'll create constants for these in the `Compiler` class.